feat: allow moving focus to floating windows with keyboard shortcuts

Daxtorim opened the pull request:
<!-- This won't be rendered!
[CHECKLIST]
- I read the contributing guide (CONTRIBUTING.md)
- My code follows the style guidelines of this project
- I have performed a self-review of my code
- I have commented on my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
-->

## Summary

Before, switching focus with keyboard shortcuts was only possible
between tiled windows. Now one can also navigate to floating windows.

When finding candidates for focus switching, floating windows will be
treated as smaller than they really are. This means that floating
windows can overlap tiled windows slightly without automatically
becoming the first choice for focus switching. This improves navigation
around floating windows as switching behavior will feel more
predictable.

## Test Plan

1. Install PR version of script
2. Open a few windows
3. Make one window floating
4. Try to navigate to that window with only keyboard shortcuts


https://user-images.githubusercontent.com/72616153/143285123-1fec66c8-4220-409f-8870-7965ec428522.mp4

gikari commented:
This needs to be rebased.

Daxtorim committed:
feat: allow moving focus to floating windows with keyboard shortcuts

Before, switching focus with keyboard shortcuts was only possible
between tiled windows. Now one can also navigate to floating windows.

When finding candidates for focus switching, floating windows will be
treated as smaller than they really are. This means that floating
windows can overlap tiled windows slightly without automatically
becoming the first choice for focus switching. This improves navigation
around floating windows as switching behavior will feel more
predictable.

Daxtorim commented:
Ready for review.

gikari reviewed (changes requested):
Nice work, but some refactoring is needed.

Daxtorim committed:
refactor: split getNeighborCandidates() up and do not return null

Daxtorim committed:
refactor: check only tiled windows for overlap when narrowing selection

This works better for multiple floating windows by jumping to the next
adjacent window instead of possibly up or down based on the last window.
